program/src/logic.rs

```diff
@@ -237,6 +252,14 @@ pub fn burn_st_sol<'a, 'b>(
         spl_token::state::Account::unpack_from_slice(&accounts.st_sol_account.data.borrow())?;
 
     // Check if the user is the account owner.
+    // LOW
+    // This is a bit of an anti-pattern with SPL token. Typically, to
+    // be safe, we encourage people to approve a certain amount of their SPL
+    // tokens to a delegate, and then sign with that.  That way, if you're
+    // interacting with a program, you know that the program can only take the
+    // amount delegated, and nothing more.  When you sign with the owner, the
+    // program could take all of your tokens. So instead of forcing a signature from
+    // the owner, you want to allow a delegate to sign.
```

Comment:
Ok, that's fine.  As an alternative, it would be easy just to have two accounts: owner and signer, where the signer is either a delegate or the owner.  Your choice!